core/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -109,7 +109,7 @@ protected Result prepare(ShardId shardId, UpdateRequest request, final GetResult
                                 request.script.getIdOrCode());
                     }
                     UpdateResponse update = new UpdateResponse(shardId, getResult.getType(), getResult.getId(),
-                            getResult.getVersion(), DocWriteResponse.Result.NOOP);
+                            getResult.getVersion(), DocWriteResponse.Result.NOOP, System.currentTimeMillis() - nowInMillis.getAsLong());
```

Comment:
`took` should always use `System.nanoTime` because it is a relative clock. `System.currentTimeMillis` can slew backwards if something like NTP is running on the system.